sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -162,10 +196,10 @@ object ModeBuilder extends ExpressionBuilder {
   override def build(funcName: String, expressions: Seq[Expression]): Expression = {
     val numArgs = expressions.length
     if (numArgs == 0) {
-      Mode(UnresolvedWithinGroup)
+      Mode(UnresolvedWithinGroup, collationEnabled = SQLConf.get.collationEnabled)
     } else if (numArgs == 1) {
       // For compatibility with function calls without WITHIN GROUP.
-      Mode(expressions(0))
+      Mode(expressions(0), collationEnabled = SQLConf.get.collationEnabled)
```

Comment:
I think we generally don't branch expression instantiation based on the "collationEnabled" flag

the idea is usually to take the collationId from the string arguments (children), and use something like `supportsBinaryEquality` to branch execution in order to preserve original performance for UTF8_BINARY

Reply:
will do!